tests/test_server.py

```diff
@@ -25,22 +19,49 @@ def negotiate_and_ntlm_auth():
     return get_auth_response('NTLM', advertise_nego_and_ntlm=True)
 
 def get_auth_response(auth_type, advertise_nego_and_ntlm=False):
-    response_headers = {'WWW-Authenticate':auth_type if not advertise_nego_and_ntlm else 'Negotiate, NTLM'}
+    # Set the default responses
+    response_headers = {'WWW-Authenticate': auth_type if not advertise_nego_and_ntlm else 'Negotiate, NTLM'}
     status_code = 401
     response = "auth with '%s\\%s':'%s'" % (domain, username, password)
 
-    # 2nd request
-    if request.headers.get('Authorization','') == REQUEST_2_TEMPLATE.format(auth_type):
-        response_headers = {'WWW-Authenticate':RESPONSE_2_TEMPLATE.format(auth_type)}
+    # Get the actual header that is returned by requests_ntlm
+    actual_header = request.headers.get('Authorization', '')
+
+    # Check what the message type is from the header
+    if actual_header == '':
+        # This is the initial connection, need to return a 401
+        message_type = None
+    else:
+        message_type = get_message_type(auth_type, actual_header)
+
+    # Validate that the message type is either a negotiate or authenticate message and act accordingly
+    if message_type == 1:
```

Comment:
It'd be nice to have named values for these magic numbers so that we know what they mean.